internal/store/selector/field.go

```diff
@@ -0,0 +1,303 @@
+package selector
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/flightctl/flightctl/internal/flterrors"
+	"github.com/flightctl/flightctl/pkg/queryparser"
+	"github.com/flightctl/flightctl/pkg/queryparser/sql"
+	"gorm.io/gorm/schema"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/selection"
+)
+
+type fieldSelector struct {
+	parser        queryparser.Parser
+	fieldResolver *selectorFieldResolver
+	operators     map[selection.Operator]string
```

Comment:
It looks like this map never changes, so why have it in this struct?

Reply:
This can be done, but keep in mind that this package will have more selectors (i.e., labels) that will use the same methods and will have their operator map. I wanted to avoid name collisions for this small map with others. 
Will appreciate your advice.

Reply:
Done